common.c

```diff
@@ -319,6 +319,11 @@ int connect_addr(struct connection *cnx, int fd_from, connect_blocking blocking)
     res = getpeername(fd_from, from.ai_addr, &from.ai_addrlen);
     CHECK_RES_RETURN(res, "getpeername", res);
 
+    if (cfg.protocols.resolve_on_forward) {
```

Comment:
I would expect this to be `if (cnx->proto->resolve_on_forward) { ` instead (I'm not sure what this does: fetch the config item of the first protocol? cfg.protocols is an array...)

Reply:
You're totally rigth... 5 min hacked together :/